src/network/network_content_gui.cpp

```diff
@@ -785,14 +785,14 @@ class NetworkContentListWindow : public Window, ContentCallback {
 				auto it = this->vscroll->GetScrolledItemFromWidget(this->content, pt.y, this, WID_NCL_MATRIX);
 				if (it == this->content.end()) return; // click out of bounds
 
-				this->selected = *it;
-				this->list_pos = it - this->content.begin();
-
 				const NWidgetBase *checkbox = this->GetWidget<NWidgetBase>(WID_NCL_CHECKBOX);
 				if (click_count > 1 || IsInsideBS(pt.x, checkbox->pos_x, checkbox->current_x)) {
-					_network_content_client.ToggleSelectedState(*this->selected);
+					_network_content_client.ToggleSelectedState(**it);
```

Comment:
Typo or am I missing something?

```suggestion
					_network_content_client.ToggleSelectedState(*it);
```

Reply:
You are missing something. `it` is the iterator. `*it` is dereferencing the iterator, which gives a `ContentInfo *`, and `**it` is dereferencing that to give a `ContentInfo`, which is what `ToggleSelectedState()` takes (as a reference.)